dev-java/gwt/gwt-2.9.0.ebuild

```diff
@@ -0,0 +1,93 @@
+# Copyright 1999-2021 Gentoo Authors
+# Distributed under the terms of the GNU General Public License v2
+
+EAPI=7
+
+JAVA_PKG_IUSE="source"
+
+inherit java-pkg-2 java-ant-2 java-utils-2
+
+GWT_TOOLS_COMMIT="194772ed94078802d8030136796de344eb1fdbe1"
+GWT_TOOLS_PN="tools"
+
+DESCRIPTION="Google Web Toolkit library"
+HOMEPAGE="http://www.gwtproject.org/"
+SRC_URI="https://github.com/gwtproject/${PN}/archive/${PV}.tar.gz -> ${P}.tar.gz
+		https://github.com/gwtproject/${GWT_TOOLS_PN}/archive/${GWT_TOOLS_COMMIT}.tar.gz -> ${PN}-${GWT_TOOLS_PN}-${GWT_TOOLS_COMMIT}.tar.gz"
+# While the test suite does compile and does execute with an exit status of 0,
+# there are many tests that end with FAILED status. It is unclear if this is
+# expected from upstream or not, but in order to keep things honest, the test
+# suite is being restricted
+RESTRICT="mirror test"
+SLOT="2.8"
+LICENSE="Apache-2.0"
+KEYWORDS="~amd64 ~x86 ~amd64-linux ~x86-linux"
+
+CDEPEND="
+	dev-java/json:0
+	dev-java/guava:20
+	dev-java/rhino:1.6
+	>=dev-java/asm-5.0.3:4
+	dev-java/eclipse-ecj:4.5
+	dev-java/validation-api:1.0
+"
+
+RDEPEND="
+	${CDEPEND}
+	>=virtual/jre-1.8:*"
+DEPEND="
+	${CDEPEND}
+	>=virtual/jdk-1.8:*
+"
+
+JAVA_ANT_REWRITE_CLASSPATH="yes"
+EANT_BUILD_TARGET="build"
+EANT_GENTOO_CLASSPATH="
+		json
+		asm-4
+		guava-20
+		rhino-1.6
+		eclipse-ecj-4.5
+		validation-api-1.0
+"
+
+PATCHES=( "${FILESDIR}/${PN}-2.9.0-jsinterop-classpath.patch" )
+
+src_prepare() {
+	default
+
+	java-pkg-2_src_prepare
+}
```

Comment:
It's really weird the java eclass(es) don't handle applying patches, but reading them this does seem correct.

Reply:
yeah, this caught me too :-)

Reply:
FYI, I've opened a bug about this behavior some time ago (https://bugs.gentoo.org/728814).
Some ebuilds had `PATCHES` set but didn't do any patching.